src/main/java/org/spongepowered/api/resource/ResourceTypes.java

```diff
@@ -0,0 +1,15 @@
+package org.spongepowered.api.resource;
+
+import org.spongepowered.api.util.generator.dummy.DummyObjectProvider;
+
+// TODO I don't like this
+public class ResourceTypes {
```

Comment:
I wouldn't add this in the API, this seems more like a implementation detail to me. The only reason to still use the `assets` package is because it's used by the Asset API, the assets package in vanilla will only be used for textures, etc. Maybe just use the `assets` package for the Asset API and the `data` package for the resource api?